src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/InstanceDiscovery.cs

```diff
@@ -104,14 +102,18 @@ internal static class InstanceDiscovery
             return entry;
         }
 
+        public static string FormatAuthorizeEndpoint(string host, string tenant)
+        {
+            return $"https://{host}/{tenant}/oauth2/authorize";
+        }
+
         // No return value. Modifies InstanceCache directly.
         private static async Task DiscoverAsync(string host, bool validateAuthority, CallState callState)
         {
-            string tentativeAuthorizeEndpoint = AuthorizeEndpointTemplate.Replace("{host}", host);
+            string tentativeAuthorizeEndpoint = FormatAuthorizeEndpoint(host, "irrelevant");
```

Comment:
Hmm I'd just create an override of this function with one parameter, rather than requiring the caller to pass garbage values.